amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/LeaseTableManager.java

```diff
@@ -0,0 +1,74 @@
+package software.amazon.kinesis.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.model.DeleteTableRequest;
+import software.amazon.awssdk.services.dynamodb.model.ListTablesRequest;
+import software.amazon.awssdk.services.dynamodb.model.ListTablesResponse;
+import software.amazon.kinesis.common.FutureUtils;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class LeaseTableManager {
+
+    private final DynamoDbAsyncClient dynamoClient;
+
+    public LeaseTableManager(DynamoDbAsyncClient dynamoClient) throws URISyntaxException, IOException {
+        this.dynamoClient = dynamoClient;
+    }
+
+    private List<String> listAllLeaseTables() throws Exception {
+        final ListTablesRequest request = ListTablesRequest.builder().build();
+        final ListTablesResponse response;
+        try {
+            response = FutureUtils.resolveOrCancelFuture(dynamoClient.listTables(request), Duration.ofSeconds(60));
```

Comment:
this is paginated call, u might not get all the tables in one go, perform paginated call